### PR TITLE
Publicly expose _LRScheduler to LRScheduler

### LINKED ISSUE #61232: [FR] rename LR scheduler base class _LRScheduler to be non-private
## 🚀 Feature
Rename `_LRScheduler` to something like `LRScheduler` without underscore. 

## Motivation

Allows for writing typed code much better. We have `torch.optim.Optimizer` but no public base class for learning rate schedulers.

### PR DESCRIPTION
**Pull request by janeyx99**
Fixes #61232

### CODE REVIEW

**Review by soulitzer — APPROVED** [review]
Cool!

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def step(self, epoch=None):
         self._last_lr = [group['lr'] for group in self.optimizer.param_groups]
 
 
+# Including _LRScheduler for backwards compatibility
+# Subclass instead of assign because we want __name__ of _LRScheduler to be _LRScheduler (assigning would make it LRScheduler).
+class _LRScheduler(LRScheduler):
+    pass
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:169]: If _LRScheduler was private in the first place, do we actually need this?

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/23957

File: torch/optim/lr_scheduler.py (hunk 1 above)
Comment (soulitzer) [reply torch/optim/lr_scheduler.py:169]: I see a good amount of internal and external usage that would break.

Comment (janeyx99) [reply torch/optim/lr_scheduler.py:169]: Right, I was surprised that it could still be imported even though it was marked with _

**Comment by janeyx99** [comment]
@vadimkantorov Thanks for bringing to attention the relevant LRScheduler issues, unfortunately, we don't have anyone focused + working on them today.
